src/modules/simulation/gz_bridge/GZBridge.cpp

```diff
@@ -211,6 +211,23 @@ int GZBridge::init()
 		return PX4_ERROR;
 	}
 
+	// Set Physics rtf
+	const char *speed_factor_str = std::getenv("PX4_SIM_SPEED_FACTOR");
+
+	if (speed_factor_str) {
+
+		double speed_factor = std::atof(speed_factor_str);
+		gz::msgs::Physics p_req;
+		p_req.set_max_step_size(speed_factor * 0.004);
+		p_req.set_real_time_factor(-1.0);
+		std::string world_physics = "/world/" + _world_name + "/set_physics";
+		std::string physics_service{world_physics};
+
+		if (!callPhysicsMsgService(physics_service, p_req)) {
+			return PX4_ERROR;
+		}
+	}
```

Comment:
It is probably harmless, but should we maybe configure the physics before spawning the model?

Reply:
I placed the physics service call further up in the init method, but not sure if this is before the model spawn you mentioned @Jaeyoung-Lim